libraries/lib-wave-track/WaveTrack.cpp

```diff
@@ -2725,7 +2738,7 @@ XMLTagHandler *WaveTrack::HandleXMLChild(const std::string_view& tag)
       auto clip = std::make_shared<WaveClip>(1,
          mpFactory, mLegacyFormat, mLegacyRate, GetWaveColorIndex());
       const auto xmlHandler = clip.get();
-      InsertClip(std::move(clip));
+      mClips.push_back(std::move(clip));
```

Comment:
I'd prefer to keep exclusively using `InsertClip` for adding clips. Here you bypass it because it's a case where it's okay that the clip is empty, but other coders later may not realize this and think it's okay to just push a new clip.
Would you please consider adding a boolean argument to `InsertClip` which defaults to rejecting empty clips ?

Reply:
Sure